doc/tutorials/active_dendrite/nestml_active_dendrite_tutorial.ipynb

```diff
@@ -0,0 +1,1162 @@
+    "## Pulse-shaped dendritic spike\n",
+    "\n",
+    "We will use a standard, linear integrate-and-fire neuron with the governing equation:\n",
+    "    \n",
+    "\\begin{align}\n",
+    "\\frac{dV_m}{dt} &= -\\frac{1}{\\tau_m} (V_m - E_L) + \\frac{1}{C_m} (I_{syn} + I_{dend\\_sp})\n",
+    "\\end{align}\n",
+    "\n",
+    "Here, the term $I_{syn}$ contains all the currents flowing into the soma due to synaptic input, and $I_{dend\\_sp}$ contains the contribution of an dendritic spike.\n",
+    "\n",
+    "### Implementing the pulse shape\n",
+    "\n",
+    "The dendritic spike is modeled here as a rectangular (pulse) function, parameterized by a height (current strength) and width (duration). \n",
+    "\n",
+    "```\n",
+    "parameters:\n",
+    "  I_dend_sp_peak pA = 150 pA   # current clamp value for I_dend during a dendritic spike\n",
+    "  T_dend_sp ms = 10 ms         # time window over which the dendritic current clamp is active\n",
+    "  ...\n",
+    "```\n",
+    "\n",
+    "We also define a synaptic current threshold that, when crossed, initiates the dendritic spike:\n",
+    "\n",
+    "```\n",
+    "parameters:\n",
+    "  I_th pA = 100 pA         # current-threshold for a dendritic spike\n",
+    "  ...\n",
+    "```\n",
+    "\n",
+    "The current is switched on and off as follows. When a dendritic spike fires, the magnitude of the ``I_dend_spike`` current is set to ``I_dend_sp_ampl``, and a timer variable ``t_dend_sp`` is set to the duration of the spike current pulse, ``T_dend_sp``. At each future run of the NESTML ``update`` block, the timer is decremented until it reaches 0, at which point the dendritic spike current ``I_dend_spike`` is set to zero, and the state of the synaptic integration is reset by setting ``I_dend`` and its derivative to zero.\n",
+    "```\n",
+    "update:\n",
+    "  if t_dend_sp > 0 ms:\n",
+    "    # during a dendritic spike pulse\n",
+    "    t_dend_sp -= resolution()\n",
+    "    if t_dend_sp <= 0 ms:\n",
+    "      # end of dendritic spike\n",
+    "      I_dend_spike = 0 pA\n",
+    "      t_dend_sp = 0 ms\n",
+    "    end\n",
+    "  end\n",
+    "\n",
+    "  if I_dend > I_th:\n",
+    "    # current-threshold, emit a dendritic spike\n",
+    "    t_dend_sp = T_dend_sp\n",
+    "    I_dend_spike = I_dend_sp_ampl\n",
+    "  end\n",
+    "```\n",
+    "\n",
+    "The complete neuron is as follows. Two other somatic currents are also defined, and could correspond to, for example, NDMA and GABA currents on ports ``I_1`` and ``I_3``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nestml_active_dend_model = '''\n",
+    "neuron iaf_psc_exp_nonlineardendrite:\n",
+    "  initial_values:\n",
+    "    V_m mV = 0 mV         # membrane potential\n",
+    "    t_dend_sp ms = 0 ms   # dendritic spike timer\n",
+    "    I_dend_sp pA = 0 pA   # dendritic spike current magnitude\n",
```

Comment:
I understand the objection. The distinction is made on technical grounds. Because a call to ``integrate_ODEs()`` evolves all of the ODEs/system state in unison, and because *I_dend* appears on the right-hand side of the differential equation for *dV_m/dt*, it could be that, during the integration of the single timestep, the value of *I_dend* drifts away from 0 (or whatever value it was set to before the call to ``integrate_ODEs()``). To prevent this, I came up with the option of splitting the contributions, and  using a multiplicative constant (*dend_curr_enabled*).